File: lib/nodes/sources/growing_list.dart
import 'list_source.dart';

class GrowingListSource extends ListSource<int> {
  final int limit;

  GrowingListSource([this.limit = 5]) : super();

  @override
  Future<void> init() async {
    await super.init();

    insertRow(0);

    for (int i = 1; i < limit; i++)
      Future.delayed(
        Duration(milliseconds: (i + 1) * 100),
        () => insertRow(i),
      );

    Future.delayed(
      Duration(milliseconds: (limit + 1) * 100),
      close,
    );
  }
}

File: src/nodes/sources/growing_list.ts

import _ from 'lodash'
import { delayed } from '../..'
import { ListSource } from './list'

export class GrowingListSource extends ListSource<number> {
    readonly limit: number
    readonly intervalInMs: number = 100

    constructor(limit = 5, autoClose = false) {
        super()
        this.limit = limit

        _.times(this.limit, (i) => {
            delayed(i * this.intervalInMs, () => {
                this.insertRow(i)
            })
        })
        if (autoClose)
            delayed((this.limit) * this.intervalInMs, this.close.bind(this))
    }
}
